be2-scala/src/main/scala/ch/epfl/pop/storage/DbActorNew.scala

```diff
@@ -37,21 +37,85 @@ case class DbActorNew(
     val channelData: ChannelData = readChannelData(channel)
     storage.write(
       (channel.toString, channelData.addMessage(message.message_id).toJsonString),
-      (s"$channel:${message.message_id}", message.toJsonString)
+      (s"$channel${Channel.SPECIAL_SEPARATOR}${message.message_id}", message.toJsonString)
```

Comment:
I hadn't realized but there's the same read-modify-write concurrency issue here.